dataset_api.md

```diff
@@ -0,0 +1,155 @@
+Incorporate the changes proposed from Martynas, with the exception of graphs(), which would now return a dictionary of graph names (URIRef or BNode) to Graph objects (as the graph's identifier would be removed).
+
+```
+add add_named_graph(uri: IdentifiedNode, graph: Graph) method
+add has_named_graph(uri: IdentifiedNode) method
+add remove_named_graph(uri: IdentifiedNode) method
+add replace_named_graph(uri: IdentifiedNode, graph: Graph)) method
+add graphs() method as an alias for contexts()
+add default_graph property as an alias for default_context
```

Comment:
Is it returning the default `Graph` or the name of the default graph? I would tend to prefer the former for easy lookup of the default graph. Also, do we want the default graph to have a graph name like currently or do we want to stop doing it?

Reply:
It would return the default graph not the name.

In the current discussion we concluded that the default graph should **not** have a name; for consistency with the spec. How achievable this is I don't know, we may be able to use `None` for the name where code/storage requires it, not sure.

Reply:
Thanks! 100% agreed this is the good behavior and the default graph shouldn't have a name